src/store.js

```diff
@@ -25,7 +26,22 @@ export default () => {
   reducerRegistry.setChangeListener(asyncReducers =>
     store.replaceReducer(combineReducers(asyncReducers))
   );
-  sagaMiddleware.run(rootSaga)
+  // We set an event listener for the saga registry
+  // So that whenever a new saga gets added
+  // We replace the sagas with the new ones
+  sagaRegistry.setChangeListener((sagas) => {
+    function* allSagas(getState) {
+        yield all(sagas);
+    }
+    sagaMiddleware.run(allSagas);
```

Comment:
you should probably call cancel on the task that `sagaMiddleware.run` returns before calling run again:
```js
let task;
.
.
.
if (task) {
  task.cancel();
}
task = sagaMiddleware.run(allSagas);
```

Reply:
Hey @sorodrigo thank you so much for the review  🙇